step/step.go

```diff
@@ -332,51 +330,21 @@ func (s XcodeTestRunner) validateXcodeVersion(input *Input, xcodeMajorVersion in
 	return nil
 }
 
-func (s XcodeTestRunner) getSimulatorForDestination(destinationSpecifier string) (simulator.Simulator, error) {
-	var sim simulator.Simulator
-	var osVersion string
-
+func (s XcodeTestRunner) getSimulatorForDestination(destinationSpecifier string) (destination.Device, error) {
 	simulatorDestination, err := destination.NewSimulator(destinationSpecifier)
-	if err != nil {
-		return simulator.Simulator{}, fmt.Errorf("invalid destination specifier (%s): %w", destinationSpecifier, err)
+	if err != nil || simulatorDestination == nil {
```

Comment:
I feel like if err == nil, `simulatorDestination` shouldn't be nil.

also if err == nil and simulatorDestination == nil the returning error will look strange, since err is nil

Reply:
Moved to a separate check.